File: api-code-examples/python/doc-list.py
import iroh
import asyncio

async def main():
    # Create in memory iroh node
    node = await iroh.IrohNode.memory()
    node_id = await node.node_id()
    print("Started Iroh node: {}".format(node_id))

    # create a document
    doc = await node.doc_create()
    print("Created doc: {}".format(doc.id()))

    # create a second document
    doc = await node.doc_create()
    print("Created doc: {}".format(doc.id()))

    # list all your documents
    docs = await node.doc_list();
    print("List all {} docs:".format(len(docs)))
    # doc ids are also called "namespace ids"
    for namespace_and_capability in docs:
        print("\t{}".format(namespace_and_capability.namespace))

# Output:
# Started Iroh node: jplmb4cgk2pxw3dwjehk7oes7ddphftlh3vdiib4e5bwhq2nnokq
# Created doc: mr42ra4f6n63kkd6kmqghgonsanqqtpv5cvkgcu7bpg2zaauimrq
# Created doc: n4w6eip5cgxlv33dl3nln544km76t4gt2etpml6kc6qpo6sqmdua
# List all 2 docs:
# 	mr42ra4f6n63kkd6kmqghgonsanqqtpv5cvkgcu7bpg2zaauimrq
# 	n4w6eip5cgxlv33dl3nln544km76t4gt2etpml6kc6qpo6sqmdua

asyncio.run(main())

File: api-code-examples/go/doc-list.go
package main

import (
	"fmt"

	"github.com/n0-computer/iroh-ffi/iroh-go/iroh"
)

func main() {
	node, err := iroh.NewIrohNode("iroh_data_dir")
	if err != nil {
		// real programs handle errors!
		panic(err)
	}

	// create one document
	doc, err := node.DocCreate()
	if err != nil {
		panic(err)
	}
	fmt.Printf("Created document %s\n", doc.Id().ToString())

	// create a second document
	doc, err = node.DocCreate()
	if err != nil {
		panic(err)
	}
	fmt.Printf("Created document %s\n", doc.Id().ToString())

	// list all your documents
	docs, err := node.DocList()
	if err != nil {
		panic(err)
	}

	fmt.Printf("Listing all %d documents:\n", len(docs))
	// doc ids are also called "NamespaceIds"
	for _, namespaceAndCapability := range docs {
		fmt.Printf("\t%s\n", namespaceAndCapability.Namespace.ToString())
	}
}
